### ISSUE TITLE
Should report error for use of generics with too few parameters

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#1418]
```python
from typing import Generic, TypeVar


T = TypeVar("T")
S = TypeVar("S")


class Foo(Generic[T, S]):
    ...


def g() -> Foo[int]: # Runtime Error, but pyright doesn't output any error
    ...
```

The above code doesn't run because `Foo[int]` raises `TypeError`, but pyright doesn't show any error.
I think it should be reported.

**Comment by erictraut** [comment]
PEP 484 indicates that missing type arguments should be assumed as `Any`, which is what Pyright does by default. So in this case, `Foo[int]` is interpreted as `Foo[int, Any]` (or more accurately `Foo[int, Unknown]`.

Pyright is able to report missing type arguments if you enable the `reportMissingTypeArgument` diagnostic check. This check is off by default in "basic" type checking mode, but you can enable it individually. It is also enabled by default in "strict" type checking mode.

**Comment by tamuhey** [comment]
Thanks, I understand it is not a bug.
But the description in PEP484 about the omission of generics parameter is ambiguous, since there is no description of a case where some of multiple parameters are omitted.

Although PEP doesn't mention, Python seems to only allow for generics parameters as "omit all" or "assign all":

```python
class Foo(Generic[T, S]):
    ...


def f() -> Foo[int, str]:  # OK
    ...


def g() -> Foo:  # OK, and PEP484 says it is equivalent to `Foo[Any]`
    ...


def h() -> Foo[int]:  # ERROR, PEP484 says nothing in this case.
    ...
```

Pyright accepts both of `Foo` and `Foo[int]`, while Python only accepts `Foo`.
If `reportMissingTypeArgument` is enabled, Pyright denies both of `Foo` and `Foo[int]`. 

I think Pyright should always report error `Foo[int]`, because it raises runtime error. (Of course, doesn't report any error for `Foo` is ok, because it doesn't raise any runtime error)

**Comment by erictraut** [comment]
That's a reasonable argument. I've made the change as you suggested. When type arguments are specified but there are either too few or two many (resulting in a runtime error), the diagnostic is reported using the reportGeneralTypeIssues diagnostic rule. If type arguments are omitted entirely, the diagnostic is reported using the reportMissingTypeArgument rule.

**Comment by erictraut** [comment]
This is included in Pyright 1.1.106, which I just published. It will also be included in the next release of Pylance.
